src/htm/algorithms/TemporalMemory.hpp

```diff
@@ -625,21 +645,34 @@ class TemporalMemory : public Serializable
   vector<SynapseIdx> numActiveConnectedSynapsesForSegment_;
   vector<SynapseIdx> numActivePotentialSynapsesForSegment_;
 
-  Real anomaly_;
-
   Random rng_;
 
+  /**
+   * holds logic and data for TM's anomaly
+   */
+  struct anomaly_tm {
+    protected:
+      friend class TemporalMemory;
+      Real anomaly_ = 0.5f; //default value
+      ANMode mode_ = ANMode::RAW;
+      AnomalyLikelihood anomalyLikelihood_; //TODO provide default/customizable params here
+  };
+
 public:
   Connections connections;
   const UInt &externalPredictiveInputs = externalPredictiveInputs_;
+
+  anomaly_tm tmAnomaly_;
   /*
    *  anomaly score computed for the current inputs
    *  (auto-updates after each call to TM::compute())
    *
    *  @return a float value from computeRawAnomalyScore()
    *  from Anomaly.hpp
    */
-  const Real &anomaly = anomaly_;
+const Real &anomaly = tmAnomaly_.anomaly_; //this is position dependant, the struct anomaly_tm must be defined before this use,
+// otherwise this surprisingly compiles, but a call to `tmAnomaly_.anomaly` segfaults!
```

Comment:
wow, Interesting ...
This seems like a problem with the C++ language, or maybe the compiler?  Thanks for making note of this issue.

Reply:
yep, strange thing, kept me busy for a while. The compiler might be, I've been using g++-9 recently due to the ParallelTS.